File: String rp or pr - GFG/string-rp-or-pr.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*; 

class GFG{
    public static void main(String args[]) throws IOException { 
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in)); 
  
        
        int t = Integer.parseInt(br.readLine()); 

        while(t > 0){
            
            String input_line1[] = br.readLine().trim().split("\\s+");
            int X = Integer.parseInt(input_line1[0]);
            int Y = Integer.parseInt(input_line1[1]);
            
            String S = br.readLine(); 
            
            Solution ob = new Solution();
		
            System.out.println(ob.solve(X,Y,S)); 

            t--;
        }
    } 
} 
// } Driver Code Ends


//User function Template for Java

class Solution 
{ 
    static long solve(int X,int Y, String S)
	{    
	    // code here
        // x//- pr(s1) , Y -rp(s2)
        String s1="pr" ; String s2="rp";
        if(X<Y)
        {
            int temp = X; X=Y; Y=temp;
	       
	       String t =s1; s1=s2; s2=t;
        }
        long ans=0;
        // //now we know X has larger cost in any case and has string associated named s1.
        Stack<Character> st = new Stack<>();
	   for(int i =S.length()-1; i>=0; i--)
	   {
	       Character curr= S.charAt(i), first=s1.charAt(0), second = s1.charAt(1);
	       if(st.size()>0 && curr ==first && st.peek()==second)
	       {
	           st.pop();
	           ans+=X;
	       }
	       else
	       st.push(curr);
	   }
	   
	    //step -2
	    StringBuilder rem = new StringBuilder();
	   while(st.size()>0)
	   rem.append(st.pop());
	   
	    S = rem.toString();
	   //in rem string check for the other cost Y, if any instances present , remove them now
	  
	   for(int i =S.length()-1; i>=0; i--)
	   {
	       Character curr= S.charAt(i), first=s2.charAt(0), second = s2.charAt(1);
	       if(st.size()>0 && curr ==first && st.peek()==second)
	       {
	           st.pop();
	           ans+=Y;
	       }
	       else
	       st.push(curr);
	   }
	   return ans;
	}
} 

File: String rp or pr - GFG/string-rp-or-pr.cpp
//{ Driver Code Starts
//Initial Template for C++
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//User function Template for C++
class Solution{   
public:
    long long solve(int X,int Y,string S){
      //code here
       // code here
	   // X - pr(String s1), Y - rp(String s2)
	    string s1="pr"; string s2="rp";
	   // decide which has greater cost
	   //if X is greater - np - proceed
	   if(X<Y)
	   {
	       int temp = X; X=Y; Y=temp; 
	       
	       string t =s1; s1=s2; s2=t;
	   }
	   
	   long long ans=0;
	   //now we know X has larger cost in any case and has string associated named a.
	   //Stack<Character> st = new Stack<>();
	   stack<char> st;
	   for(int i =S.size()-1; i>=0; i--)
	   {
	     char curr=S[i], first = s1[0], second= s1[1];
	       if(st.size()>0 && curr==first && st.top()==second)
	       {
	           st.pop();
	           ans+=X;
	       }
	       else
	       st.push(curr);
	   }
	   
	  S="";
	   while(st.size()>0)
	  {
	      S.push_back(st.top());
	      st.pop();
	  }
	   
	
	   //in rem string check for the other cost Y, if any instances present , remove them now
	   for(int i =S.size()-1; i>=0; i--)
	   {
	     char curr=S[i], first = s2[0], second= s2[1];
	       if(st.size()>0 && curr==first && st.top()==second)
	       {
	           st.pop();
	           ans+=Y;
	       }
	       else
	       st.push(curr);
	   }
	   return ans;
    }
};

//{ Driver Code Starts.
signed main()
{
  int t;
  cin>>t;
  while(t--)
  {
      int X,Y;
      cin>>X>>Y;
      string S;
      cin>>S;
      Solution obj;
      long long answer=obj.solve(X,Y,S);
      cout<<answer<<endl;
  }
}
// } Driver Code Ends